### ISSUE TITLE
`no-unnecessary-polyfills`: respect `.browserslistrc`

### ISSUE THREAD

**Issue by RebeccaStevens** [sindresorhus/eslint-plugin-unicorn#2389]
### Description

Currently it will respect a `browserlist` field in `package.json` but it would be great if it could also respect the `.browserslistrc` file.

This will be a little more complicated though as that file can define multiple targets, so an option as to which target to respect would also need to be added, defaulting to `production`.

### Fail

```properties
# .browserslistrc

[production]
since 2024

[development]
last 1 version
```

```ts
if (!("groupBy" in Object)) {
  import("object.groupby").then(({ shim }) => shim())
}
```

### Pass

```properties
# .browserslistrc

[production]
since 2020

[development]
last 1 version
```

```ts
if (!("groupBy" in Object)) {
  import("object.groupby").then(({ shim }) => shim())
}
```
